File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let answer = [];
    let S = brown + yellow
    let sArr = []
    for(let h = 1; h <= S; h++) {
        let w = S/h
        if(Number.isInteger(w) && w >= h) sArr.push([w, h])
    }
    sArr.map((s) => {  
        if((s[0] - 2) * (s[1] - 2) === yellow) answer = s }
    )
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow

    for height in range(1, int(yellow ** 0.5) + 1):
        if yellow % height == 0:
            width = yellow // height
            if (width + 2) * (height + 2) == total:
                return [width + 2, height + 2] 